### PR TITLE
Added delete operation incl. according REST API

### PR DESCRIPTION
**Pull request by fooker**
Required by http://issues.opennms.org/browse/HZN-388.

### CODE REVIEW

File: cassandra/storage/src/main/java/org/opennms/newts/persistence/cassandra/CassandraSampleRepository.java [hunk 1]
```diff
@@ -249,6 +261,35 @@ public void insert(Collection<Sample> samples, boolean calculateTimeToLive) {
 
     }
 
+    @Override
+    public void delete(Context context, Resource resource) {
+        final Timestamp start = Timestamp.now().minus(m_ttl, TimeUnit.SECONDS);
```
Comment (j-white) [inline cassandra/storage/src/main/java/org/opennms/newts/persistence/cassandra/CassandraSampleRepository.java]: m_ttl can be 0 (when rows never expire), what's the expected behavior in this case?

File: cassandra/search/src/main/java/org/opennms/newts/cassandra/search/CassandraIndexer.java [hunk 2]
```diff
@@ -106,6 +107,28 @@ public void update(Collection<Sample> samples) {
 
     }
 
+    @Override
+    public void delete(final Context context, final Resource resource) {
+        final Timer.Context ctx = m_updateTimer.time();
```
Comment (j-white) [inline cassandra/search/src/main/java/org/opennms/newts/cassandra/search/CassandraIndexer.java]: This should use a different timer.

**Comment by j-white** [comment]
I've added a few comments, otherwise it looks good.

Can we add a few test cases that validate the deletes?

File: cassandra/storage/src/main/java/org/opennms/newts/persistence/cassandra/CassandraSampleRepository.java (hunk 1 above)
Comment (fooker) [inline cassandra/storage/src/main/java/org/opennms/newts/persistence/cassandra/CassandraSampleRepository.java]: I'm just thinking about the final scene of fight club...

Comment (fooker) [inline cassandra/storage/src/main/java/org/opennms/newts/persistence/cassandra/CassandraSampleRepository.java]: To be honest, I don't know.

**Comment by fooker** [comment]
I added a test case for deletion.

**Pull request merged**